File: GameDev/Editor.cs
using System;
using System.Threading.Tasks;

namespace GameDev
{
    internal interface EditorBase
    {
        void cleanup();
        Task<bool> doc_close();
    }

    internal interface Editor : EditorBase
    {
        Task<string> doc_save_as();
        Task doc_save();                
        Task doc_fresh();

        void undo();
        void redo();
        void comment();
        void upper();
        void lower();
        void find();
        void findnext();
        void findprev();
        void replace();
        void gotoline();

    }
}

File: QGameDev/Editor.cpp
#include "Editor.h"

EditorBase::EditorBase(QWidget* parent) : QWidget(parent)
{

}
EditorBase::~EditorBase()
{

}

void EditorBase::cleanup()
{

}

void EditorBase::doc_close()
{
	emit doc_close_ret(true);
}

Editor::Editor(QWidget* parent): EditorBase(parent)
{

}

Editor::~Editor()
{

}